Terminal.Gui/Core/ConsoleDriver.cs

```diff
@@ -246,109 +370,109 @@ Attribute SetAttribute (Attribute attribute, [CallerMemberName] string callerMem
 			case "TopLevel":
```

Comment:
This whole switch statement scares me. So much brute-force code! It makes me wonder if there's a way of re-factoring it as part of this PR to use reflection or something to radically simplify it?